### ISSUE TITLE
match error

### ISSUE THREAD

**Issue by eauxxx** [cijber/domainmatcher#1]
[document](https://docs.rs/eater_domainmatcher/0.1.1/eater_domainmatcher/index.html) say: 
`*.domain.tld` should match `domain.tld`
but is not.
```rust
#[test]
fn test_matcher() {
    let a = DomainPattern::<'.'>::parse("*.domain.tld").unwrap();
    assert!(a.matches("domain.tld"));
}
```
### panic
```
thread 'test_matcher' panicked at 'assertion failed: a.matches(\"domain.tld\")'
```

**Comment by the-eater** [comment]
TL;DR: It's fixed in 0.1.2

well that's awkward! I published 0.1.2 which works as prescribed and has them actually added as test cases!

https://github.com/cijber/domainmatcher/blob/main/src/lib.rs#L364

Now they all work as expected, not sure if you're interested in the cause why it was failing

But basically we create a list of "matchers", so e.g. `*.domain.tld` becomes `[wildcard, "domain", "tld"]`

and we have a "stack" of pointers which point to a location to this list, this stack starts with 1 item at index 0

so whenever it goes over a label in a domain (that's what individual parts of the domain are called), it'll go over every item in the stack, and see if the matcher at located pointer matches, if so, it'll look at the next item, and if the next item is a wildcard, it'll also add a pointer to the one after, this allows us to skip the wildcards.

Now what I forgot however, is to run this wildcard "blow up" also at the start of the algorithm, so `[wildcard, "domain", "tld"]` could never get to the end, because it never was allowed to skip the first wildcard, now I have added the wildcard blow up algorithm before it actually starts walking the domain, so now it will start with a stack that looks like `[0, 1]` so it starts either checking with the wildcard and `"domain"`

You can see that here https://github.com/cijber/domainmatcher/commit/1f9fad99f1cf18f7a00d9e09a1c12f3cbcae24ae

Sorry for the inconvenience, but hope this helps!
